### DISCUSSION TITLE
A question about positioning the cursor at the mouse click without using the Alt key

### DISCUSSION THREAD

**Discussion by New-Lanke** [sourcelair/xterm.js#4609] [General]
I want the cursor to be positioned at the mouse click without using the Alt key. Does Xterm currently support binding custom mouse events and calling related methods for moving the cursor?
If there are other ways to achieve it, please let me know. Thank you.

**Comment by jerch** [comment]
Nope, there is no exposed ready-to-go API for that, thus you are on your own here (caution, uses private stuff):
```javascript
// within an encoding you can do your cursor jumper logic
// here: move the cursor with CUP
term._core.coreMouseService.addEncoding('my_cursor_jumper', e => term.write(`\x1b[${e.row};${e.col}H`));

// to activate it:
term._core.coreMouseService.activeEncoding = 'my_cursor_jumper';
term._core.coreMouseService.activeProtocol = 'X10';
// --> now a click on a terminal cell should move the cursor there
```

X10 grabs only mouse clicks, for other mouse event types see https://github.com/xtermjs/xterm.js/blob/1c98037eede9b6cb9610e20372ec43b6b19e69b7/src/common/services/CoreMouseService.ts#L13

**Reply by New-Lanke** [reply]
After clicking to move the cursor, the input will still be attached to the end (I used the xterm-readline addon, and after pressing enter, I can get the input.). But Alt+Left Click works, the input was correctly inserted into the mouse click position.

**Reply by jerch** [reply]
Idk the xterm-readline addon, it prolly holds the buffer state itself and then ofc - just moving the cursor around in the terminal does not change that state. Maybe it has an API to jump the cursor around, then you can add it in the handler above.
